### PR TITLE
[website] new API migration

### PR DESCRIPTION
**Pull request by mba-odoo**
Task: https://www.odoo.com/web#id=12131&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.eLvFy2wnfbAiJdxS

### CODE REVIEW

File: addons/website/controllers/main.py [hunk 1]
```diff
@@ -98,22 +98,21 @@ def robots(self):
 
     @http.route('/sitemap.xml', type='http', auth="public", website=True)
     def sitemap_xml_index(self):
-        cr, uid, context = request.cr, openerp.SUPERUSER_ID, request.context
-        ira = request.registry['ir.attachment']
-        iuv = request.registry['ir.ui.view']
-        mimetype ='application/xml;charset=utf-8'
+        Ira = request.env['ir.attachment'].sudo()
```
Comment (rim-odoo) [inline addons/website/controllers/main.py]: Ira -> Attachment will be more explicit

File: addons/website/controllers/main.py [hunk 2]
```diff
@@ -98,22 +98,21 @@ def robots(self):
...
+        Iuv = request.env['ir.ui.view'].sudo()
```
Comment (rim-odoo) [inline addons/website/controllers/main.py]: Iuv -> View

File: addons/website/controllers/main.py (hunk 1 above)
Comment (rim-odoo) [inline addons/website/controllers/main.py]: Why sudo ?

File: addons/website/controllers/main.py (hunk 2 above)
Comment (rim-odoo) [inline addons/website/controllers/main.py]: Why sudo ?

File: addons/website/data/data.xml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
+<odoo>
     <data noupdate="1">
         <record id="default_website" model="website">
             <field name="name">www.openerp.com</field>
```
Comment (rim-odoo) [inline addons/website/data/data.xml]: odoo.com  ? ;-)

File: addons/website/models/ir_actions.py [hunk 4]
```diff
@@ -2,64 +2,60 @@
...
     """ Add website option in server actions. """
     _name = 'ir.actions.server'
     _inherit = ['ir.actions.server']
 
-    def _compute_website_url(self, cr, uid, id, website_path, xml_id, context=None):
-        base_url = self.pool['ir.config_parameter'].get_param(cr, uid, 'web.base.url', context=context)
-        link = website_path or xml_id or (id and '%d' % id) or ''
+    @api.multi
+    def _compute_website_url(self):
+        self.ensure_one()
```
Comment (rim-odoo) [inline addons/website/models/ir_actions.py]: why ensure one ? couldn't you compute this field for more than one server action?

File: addons/website/models/ir_actions.py [hunk 5]
```diff
@@ -2,64 +2,60 @@
...
+                 'set to is False the action cannot be run through the website.')
 
-    def on_change_website_path(self, cr, uid, ids, website_path, xml_id, context=None):
-        values = {
-            'website_url': self._compute_website_url(cr, uid, ids and ids[0] or None, website_path, xml_id, context=context)
-        }
-        return {'value': values}
+    @api.onchange('website_path')
+    def on_change_website_path(self):
+        self.website_url = self._compute_website_url()
```
Comment (rim-odoo) [inline addons/website/models/ir_actions.py]: not sure it is still needed with new computed fields

File: addons/website/models/ir_http.py [hunk 6]
```diff
@@ -32,18 +34,24 @@ class ir_http(orm.AbstractModel):
...
     def _get_converters(self):
         return dict(
-            super(ir_http, self)._get_converters(),
+            super(IrHttp, self)._get_converters(),
             model=ModelConverter,
             page=PageConverter,
         )
 
+    @api.v7
+    def _auth_method_public(self):
```
Comment (rim-odoo) [inline addons/website/models/ir_http.py]: strange signature

File: addons/website/models/ir_http.py [hunk 7]
```diff
@@ -32,18 +34,24 @@ class ir_http(orm.AbstractModel):
...
+        new_obj = self.browse(request.cr, request.uid, [])
```
Comment (rim-odoo) [inline addons/website/models/ir_http.py]: no context?

File: addons/website/controllers/main.py (hunk 1 above)
Comment (mba-odoo) [inline addons/website/controllers/main.py]: hello @rim-odoo , Here You can see on line : https://github.com/odoo/odoo/pull/8230/files#diff-167e2f4c21937f0a924ebe41ed38b16dL101 simply uid is SUPERUSER_ID so used sudo.

File: addons/website/models/ir_actions.py (hunk 5 above)
Comment (mba-odoo) [inline addons/website/models/ir_actions.py]: hello @rim-odoo , not needed  so just removed but it's going to change website_url on save,  not instant effect like on_change. Thanks

File: addons/website/controllers/main.py [hunk 8]
```diff
@@ -98,25 +98,24 @@ def robots(self):
...
-            ), context=context)
+            ))
 
-        sitemap = ira.search_read(cr, uid, [('url', '=' , '/sitemap.xml'), ('type', '=', 'binary')], ('datas', 'create_date'), context=context)
+        sitemap = Attachment.search_read([('url', '=', '/sitemap.xml'), ('type', '=', 'binary')], ('datas', 'create_date'))
         if sitemap:
             # Check if stored version is still valid
-            server_format = openerp.tools.misc.DEFAULT_SERVER_DATETIME_FORMAT
+            server_format = odoo.tools.misc.DEFAULT_SERVER_DATETIME_FORMAT
             create_date = datetime.datetime.strptime(sitemap[0]['create_date'], server_format)
```
Comment (hmo-odoo) [inline addons/website/controllers/main.py]: use fields.Datetime.from_string()

File: addons/website/controllers/main.py [hunk 9]
```diff
@@ -124,9 +123,9 @@ def create_sitemap(url, content):
 
         if not content:
             # Remove all sitemaps in ir.attachments as we're going to regenerated them
-            sitemap_ids = ira.search(cr, uid, [('url', '=like' , '/sitemap%.xml'), ('type', '=', 'binary')], context=context)
-            if sitemap_ids:
-                ira.unlink(cr, uid, sitemap_ids, context=context)
+            sitemaps = Attachment.search([('url', '=like', '/sitemap%.xml'), ('type', '=', 'binary')])
+            if sitemaps:
```
Comment (hmo-odoo) [inline addons/website/controllers/main.py]: remove if statement

File: addons/website/controllers/main.py [hunk 10]
```diff
@@ -278,11 +270,9 @@ def seo_suggest(self, keywords=None, lang=None):
 
     def get_view_ids(self, xml_ids):
```
Comment (hmo-odoo) [inline addons/website/controllers/main.py]: can we return browse records instead of ids ?

File: addons/website/controllers/main.py [hunk 11]
```diff
@@ -339,16 +326,15 @@ def multi_render(self, ids_or_xml_ids, values=None):
...
     def actions_server(self, path_or_xml_id_or_id, **post):
-        cr, uid, context = request.cr, request.uid, request.context
         res, action_id, action = None, None, None
-        ServerActions = request.registry['ir.actions.server']
+        ServerActions = request.env['ir.actions.server']
 
         # find the action_id: either an xml_id, the path, or an ID
         if isinstance(path_or_xml_id_or_id, basestring) and '.' in path_or_xml_id_or_id:
-            action_id = request.registry['ir.model.data'].xmlid_to_res_id(request.cr, request.uid, path_or_xml_id_or_id, raise_if_not_found=False)
+            action_id = request.env.ref(path_or_xml_id_or_id, raise_if_not_found=False).id
```
Comment (hmo-odoo) [inline addons/website/controllers/main.py]: use browse record.

after that, probability to remove code from line:337 to line:348

File: addons/website/models/ir_actions.py [hunk 12]
```diff
@@ -2,64 +2,56 @@
...
     """ Add website option in server actions. """
     _name = 'ir.actions.server'
     _inherit = ['ir.actions.server']
 
-    def _compute_website_url(self, cr, uid, id, website_path, xml_id, context=None):
-        base_url = self.pool['ir.config_parameter'].get_param(cr, uid, 'web.base.url', context=context)
-        link = website_path or xml_id or (id and '%d' % id) or ''
+    @api.multi
+    def _compute_website_url(self):
+        self.ensure_one()
```
Comment (hmo-odoo) [inline addons/website/models/ir_actions.py]: do not use ensure_one in compute function

File: addons/website/models/ir_actions.py [hunk 13]
```diff
@@ -2,64 +2,56 @@
...
+class ActionsServer(models.Model):
     """ Add website option in server actions. """
     _name = 'ir.actions.server'
     _inherit = ['ir.actions.server']
 
-    def _compute_website_url(self, cr, uid, id, website_path, xml_id, context=None):
-        base_url = self.pool['ir.config_parameter'].get_param(cr, uid, 'web.base.url', context=context)
-        link = website_path or xml_id or (id and '%d' % id) or ''
+    @api.multi
+    def _compute_website_url(self):
```
Comment (hmo-odoo) [inline addons/website/models/ir_actions.py]: declare compute function after fields declaration

File: addons/website/models/ir_http.py [hunk 14]
```diff
@@ -32,18 +34,19 @@ class ir_http(orm.AbstractModel):
...
-            super(ir_http, self)._get_converters(),
+            super(IrHttp, self)._get_converters(),
             model=ModelConverter,
             page=PageConverter,
         )
 
     def _auth_method_public(self):
         if not request.session.uid:
-            website = self.pool['website'].get_current_website(request.cr, openerp.SUPERUSER_ID, context=request.context)
+            self = self.browse(request.cr, request.uid, [], request.context)
```
Comment (hmo-odoo) [inline addons/website/models/ir_http.py]: why ?

File: addons/website/models/ir_ui_view.py [hunk 15]
```diff
@@ -140,46 +134,40 @@ def render(self, cr, uid, id_or_xml_id, values=None, engine='ir.qweb', context=N
...
-
-    def _prepare_qcontext(self, cr, uid, context=None):
-        if not context:
-            context = {}
+        return super(View, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
 
-        company = self.pool['res.company'].browse(cr, SUPERUSER_ID, request.website.company_id.id, context=context)
+    @api.model
+    def _prepare_qcontext(self):
+        company = self.env['res.company'].sudo().browse(request.website.company_id.id)
```
Comment (hmo-odoo) [inline addons/website/models/ir_ui_view.py]: no need to browse again

File: addons/website/models/ir_ui_view.py [hunk 16]
```diff
@@ -140,46 +134,40 @@ def render(self, cr, uid, id_or_xml_id, values=None, engine='ir.qweb', context=N
...
 
         editable = request.website.is_publisher()
-        translatable = editable and context.get('lang') != request.website.default_lang_code
+        translatable = editable and self.env.context.get('lang') != request.website.default_lang_code
         editable = not translatable and editable
 
         qcontext = dict(
-            context.copy(),
+            self.env.context.copy(),
```
Comment (hmo-odoo) [inline addons/website/models/ir_ui_view.py]: why need to copy  context ?

File: addons/website/models/ir_ui_view.py (hunk 15 above)
Comment (rim-odoo) [inline addons/website/models/ir_ui_view.py]: why are you doing this? why? I want answers!

Comment (mba-odoo) [inline addons/website/models/ir_ui_view.py]: @rim-odoo ,
As old code it's browsed again with SUPERUSER_ID link : https://github.com/odoo/odoo/pull/8230/files#diff-bf8d854179fb58216be065c91d78d2b6L149 but it's my bad to browse again as before. i am removing it.

Comment (rim-odoo) [inline addons/website/models/ir_ui_view.py]: ok ok... if you really need superuser, you can do something like self.sudo().request.website.company_id :)

File: addons/website/models/res_config.py [hunk 17]
```diff
@@ -1,51 +1,46 @@
...
+    social_googleplus = fields.Char(related='website_id.social_googleplus', string='Google+ Account')
+    compress_html = fields.Boolean(related='website_id.compress_html', string='Compress rendered HTML for a better Google PageSpeed result')
+    cdn_activated = fields.Boolean(related='website_id.cdn_activated', string='Use a Content Delivery Network (CDN)')
+    cdn_url = fields.Char(related='website_id.cdn_url', string='CDN Base URL')
+    cdn_filters = fields.Text(related='website_id.cdn_filters', string='CDN Filters')
+    module_website_form_editor = fields.Boolean(string="Website form builder")
+    module_website_version = fields.Boolean(string="Website A/B testing and versioning")
+
+    @api.onchange('website_id')
+    def on_change_website_id(self):
```
Comment (hmo-odoo) [inline addons/website/models/res_config.py]: why need this stuff ?

File: addons/website/models/res_config.py [hunk 18]
```diff
@@ -1,51 +1,46 @@
...
-    def create(self, cr, uid, vals, context=None):
-        config_id = super(website_config_settings, self).create(cr, uid, vals, context=context)
-        self.write(cr, uid, config_id, vals, context=context)
-        return config_id
-
-    _defaults = {
-        'website_id': lambda self,cr,uid,c: self.pool.get('website').search(cr, uid, [], context=c)[0],
-    }
+    @api.model
+    def create(self, vals):
```
Comment (hmo-odoo) [inline addons/website/models/res_config.py]: why need to override create method to write same vals ?

File: addons/website/models/website.py [hunk 19]
```diff
@@ -139,60 +136,51 @@ def unslug(s):
...
+    cdn_filters = fields.Text('CDN Filters', help="URL matching those filters will be rewritten using the CDN Base URL", default='\n'.join(DEFAULT_CDN_FILTERS))
+    partner_id = fields.Many2one(related='user_id.partner_id', relation='res.partner', string='Public Partner')
+    menu_id = fields.Many2one('website.menu', compute='_get_menu', string='Main Menu')
+
+    @api.multi
+    def _get_menu(self):
+        Menu = self.env['website.menu']
+        for res in self:
+            menus = Menu.search([('parent_id', '=', False), ('website_id', '=', res.id)], order='id', limit=1)
+            res.menu_id = menus and menus.id or False
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: just menu.id

File: addons/website/models/website.py [hunk 20]
```diff
@@ -220,44 +207,39 @@ def new_page(self, cr, uid, name, template='website.default_page', ispage=True,
         })
         return page_xmlid
 
-    def key_to_view_id(self, cr, uid, view_id, context=None):
-        View = self.pool.get('ir.ui.view')
-        return View.search(cr, uid, [
+    @api.model
+    def key_to_view_id(self, view_id):
+        return self.env['ir.ui.view'].search([
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: use limit=1

File: addons/website/models/website.py [hunk 21]
```diff
@@ -315,25 +295,29 @@ def page_search_dependencies(self, cr, uid, view_id=False, context=None):
...
+            return self.env["ir.model.data"].xmlid_to_res_model_res_id(module+'.'+name)
         except:
             return False
 
-    @openerp.tools.ormcache('id')
-    def _get_languages(self, cr, uid, id, context=None):
-        website = self.browse(cr, uid, id)
-        return [(lg.code, lg.name) for lg in website.language_ids]
+    @api.multi
+    @tools.ormcache('self.id')
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: is it really working ? why pass self.id as a key in ormcache ?

File: addons/website/models/website.py [hunk 22]
```diff
@@ -370,7 +356,7 @@ def get_alternate_languages(self, cr, uid, ids, req=None, context=None):
                 lang['hreflang'] = lang['short']
         return langs
 
-    @openerp.tools.ormcache('domain_name')
+    @tools.ormcache('domain_name')
     def _get_current_website_id(self, cr, uid, domain_name, context=None):
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: any reason for unmigrated  code ?

File: addons/website/models/website.py [hunk 23]
```diff
@@ -552,50 +544,47 @@ def search_pages(self, cr, uid, ids, needle=None, limit=None, context=None):
...
-        'child_id': fields.one2many('website.menu', 'parent_id', string='Child Menus'),
-        'parent_left': fields.integer('Parent Left', select=True),
-        'parent_right': fields.integer('Parent Right', select=True),
-    }
-
-    def __defaults_sequence(self, cr, uid, context):
-        menu = self.search_read(cr, uid, [(1,"=",1)], ["sequence"], limit=1, order="sequence DESC", context=context)
+
+    def __defaults_sequence(self):
+        menu = self.search_read([(1, "=", 1)], ["sequence"], limit=1, order="sequence DESC")
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: use self.search with domain []

File: addons/website/models/website.py [hunk 24]
```diff
@@ -552,50 +544,47 @@ def search_pages(self, cr, uid, ids, needle=None, limit=None, context=None):
...
         return menu and menu[0]["sequence"] or 0
 
-    _defaults = {
-        'url': '',
-        'sequence': __defaults_sequence,
-        'new_window': False,
-    }
+    name = fields.Char('Menu', required=True, translate=True)
+    url = fields.Char(default='')
+    new_window = fields.Boolean('New Window', default=False)
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: no need to set False value as a default for boolean field

File: addons/website/models/website.py [hunk 25]
```diff
@@ -624,125 +614,123 @@ def replace_id(old_id, new_id):
                     menu['parent_id'] = new_id
         to_delete = data['to_delete']
         if to_delete:
-            self.unlink(cr, uid, to_delete, context=context)
+            self.brwose(to_delete).unlink()
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: typo

File: addons/website/models/website.py [hunk 26]
```diff
@@ -624,125 +614,123 @@ def replace_id(old_id, new_id):
...
+    website_url = fields.Char(related="local_url", string="Attachment URL", deprecated=True) # related for backward compatibility with saas-6
 
-class res_partner(osv.osv):
+class ResPartner(models.Model):
     _inherit = "res.partner"
 
-    def google_map_img(self, cr, uid, ids, zoom=8, width=298, height=298, context=None):
-        partner = self.browse(cr, uid, ids[0], context=context)
+    @api.multi
+    def google_map_img(self, zoom=8, width=298, height=298):
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: put ensure_one()

File: addons/website/models/website.py [hunk 27]
```diff
@@ -624,125 +614,123 @@ def replace_id(old_id, new_id):
...
         params = {
-            'center': '%s, %s %s, %s' % (partner.street or '', partner.city or '', partner.zip or '', partner.country_id and partner.country_id.name_get()[0][1] or ''),
+            'center': '%s, %s %s, %s' % (self.street or '', self.city or '', self.zip or '', self.country_id and self.country_id.display_name or ''),
             'size': "%sx%s" % (height, width),
             'zoom': zoom,
             'sensor': 'false',
         }
-        return urlplus('//maps.googleapis.com/maps/api/staticmap' , params)
+        return urlplus('//osvmaps.googleapis.com/maps/api/staticmap' , params)
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: osvmaps ? is it typo ?

File: addons/website/models/website.py [hunk 28]
```diff
@@ -624,125 +614,123 @@ def replace_id(old_id, new_id):
...
+        return dict.fromkeys(self.ids, '#')
 
-    def website_publish_button(self, cr, uid, ids, context=None):
-        for i in self.browse(cr, uid, ids, context):
-            if self.pool['res.users'].has_group(cr, uid, 'base.group_website_publisher') and i.website_url != '#':
-                return self.open_website_url(cr, uid, ids, context)
+    @api.multi
+    def website_publish_button(self):
+        for i in self:
+            if self.env['res.users'].has_group('base.group_website_publisher') and i.website_url != '#':
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: check publisher user or not before for loop

File: addons/website/models/website.py [hunk 29]
```diff
@@ -624,125 +614,123 @@ def replace_id(old_id, new_id):
...
+                return self.open_website_url()
```
Comment (hmo-odoo) [inline addons/website/models/website.py]: why return inside loop ?

File: addons/website/models/res_config.py (hunk 17 above)
Comment (mba-odoo) [inline addons/website/models/res_config.py]: Hello @hmo-odoo ,
it's used onchange of website_id(many2one with website model) need to set all fields of website_config_setting and there might be the field exist or may be many2one or simple so need to set all the fields value based on many2one of website_id so need this stuff.

File: addons/website/models/res_config.py (hunk 18 above)
Comment (mba-odoo) [inline addons/website/models/res_config.py]: hello @hmo-odoo , in old api create does not forward the values to the related many2one. Write does but now in new api it's working fine so just removed create. thanks.

File: addons/website/controllers/main.py (hunk 10 above)
Comment (mba-odoo) [inline addons/website/controllers/main.py]: yes, we can, returned recordset. done

File: addons/website/models/website.py (hunk 20 above)
Comment (mba-odoo) [inline addons/website/models/website.py]: done.

File: addons/website/models/website.py (hunk 26 above)
Comment (mba-odoo) [inline addons/website/models/website.py]: done

File: addons/website/models/website.py (hunk 21 above)
Comment (mba-odoo) [inline addons/website/models/website.py]: yes it's working fine. it's update lang of cache on change of website id. Thanks

**Pull request closed without merging**